src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super();
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super();
+  }
+
+  bite(animal) {
+    if (!(animal instanceof Carnivore) && animal.hidden !== true) {
```

Comment:
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
```
I think there is no need to revert the condition here. 
It's recommended to check for truly values for better readability.
Also, you can shorten the condition with `not` for boolean